src/server/system_services/account_server.js

```diff
@@ -658,6 +658,50 @@ function delete_account(req) {
         );
 }
 
+
+/**
+ *
+ * DELETE_ACCOUNT BY PROPERTY
+ *
+ */
+ function delete_account_by_property(req) {
+    let roles_to_delete = [];
+    let accounts_to_delete = system_store.get_accounts_by_nsfs_account_config(req.rpc_params.nsfs_account_config)
+        .map(account_to_delete => {
+            if (!account_to_delete) {
+                throw new RpcError('NO_SUCH_ACCOUNT', 'No such account email: ' + req.rpc_params.email);
+            }
+            if (account_to_delete.is_support) {
+                throw new RpcError('BAD_REQUEST', 'Cannot delete support account');
+            }
+            if (String(account_to_delete._id) === String(req.system.owner._id)) {
```

Comment:
Can the _id be from a different type? 
Why do we cast?

Reply:
the case here is taken from the original delete_account function, it's done because its type is objectId and comparing between 2 objectIDs should be done by using the mongo equal function / casting to strings.